server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -635,20 +635,19 @@ public VirtConsumerMap getHostConsumersMap(Owner owner,
         if (hypervisorIds == null || hypervisorIds.isEmpty()) {
             return new LinkedList<Consumer>();
         }
-        return currentSession().createCriteria(Consumer.class)
-            .createAlias("owner", "o")
-            .add(Restrictions.eq("o.key", ownerKey))
-            .createAlias("hypervisorId", "hvsr")
-            .add(getHypervisorIdRestriction(hypervisorIds))
-            .list();
-    }
-
-    private Criterion getHypervisorIdRestriction(Collection<String> hypervisorIds) {
-        List<Criterion> ors = new LinkedList<Criterion>();
+        LinkedList<String>  hypervisorIdsUpper = new LinkedList<String>();
         for (String hypervisorId : hypervisorIds) {
-            ors.add(Restrictions.eq("hvsr.hypervisorId", hypervisorId).ignoreCase());
-        }
-        return Restrictions.or(ors.toArray(new Criterion[0]));
+            hypervisorIdsUpper.add(hypervisorId.toUpperCase());
+        }
+        String query = "SELECT c FROM Consumer c WHERE " +
+                       "(c.owner.key = :ownerKey AND UPPER(c.hypervisorId.hypervisorId) IN (:hypervisorIds)) " +
```

Comment:
We have to be careful with IN operations. Is it possible we'll hit the operator limit here?